span.go

```diff
@@ -189,16 +192,17 @@ func (r *spanS) Tracer() ot.Tracer {
 // to Instana log spans and sends them to the agent
 func (r *spanS) sendOpenTracingLogRecords() {
 	for _, lr := range r.Logs {
-		for _, lf := range lr.Fields {
-			if isErrorLogField(lf) {
-				r.sendOpenTracingLogRecord(lr)
-				break
-			}
-		}
+		r.sendOpenTracingLogRecord(lr)
 	}
 }
 
 func (r *spanS) sendOpenTracingLogRecord(lr ot.LogRecord) {
+	lvl := openTracingLogRecordLevel(lr)
+
+	if lvl > minSpanLogLevel {
```

Comment:
@andrewslotin What do you think: based on the description of this PR, does it makes sense instead of returning here to do filtering based on the minimal level we want to return?

Reply:
I'm not sure I follow, could you elaborate what did you meant by "filtering based on the minimal level we want to return"?

Reply:
Let me clarify it a little bit 😃 

If I follow logic correct: we want to send logs with level warn or error. 

I see in code we iterate over `lr.Fields` to find what is the highest log level is in the `LogRecord`. After, if this level is higher than `WarnLevel` we do not send this record.

From the code perspective, it seems there might be multiple fields within `lr` with different levels. With the current implementation, if any has a level higher than `warn` whole `LogRecord` won't be sent. 

So by filtering, I mean ignoring (filtering out) only fields where `lf.Key` is not an `error` or `warn`, but not whole the `LogRecord`. 

Does it make sense?

Reply:
Okay, I see your point now. The thing is that the logic here is supposed to be quite the opposite: if we find at least one field with key `error` or `warn`, the whole log record is considered to be of this level and thereby needs to be sent to the agent.

The problem is that log levels in the `logger` package are represented by integer numbers with `0` being the highest level `ERROR`, so we need to invert the comparison to figure out whether we should discard this record or not. I was thinking of introducing a separate type for log levels here with a different set of values, so that `ERROR` would be greater than `WARN`, but found it to be unnecessary.

Do you think it would help if I add a comment here?

Reply:
Ok, got it. I think a tiny comment will be helpful there.

Reply:
I ended up with adding `(logger.Level).Less()`, does this code make more sense now?